orca-core/src/main/java/com/netflix/spinnaker/orca/config/UserConfiguredUrlRestrictions.java

```diff
@@ -112,6 +177,29 @@ public UserConfiguredUrlRestrictions(
             ? Collections.emptySet()
             : Collections.unmodifiableSet(new HashSet<>(rejectedIps));
     this.clientProperties = clientProperties;
+    this.excludedPatterns = excludedPatterns;
+  }
+
+  InetAddress resolveHost(String host) throws UnknownHostException {
+    return InetAddress.getByName(host);
+  }
+
+  boolean isLocalhost(InetAddress addr) throws SocketException {
+    return addr.isLoopbackAddress()
+        || Optional.ofNullable(NetworkInterface.getByInetAddress(addr)).isPresent();
+  }
+
+  boolean isLinkLocal(InetAddress addr) {
+    return addr.isLinkLocalAddress();
+  }
```

Comment:
PERSONALLY not a fan of one line function wrappers like this... I'd rather have the link local direct, as it's a bit clearer to me...

Reply:
This allows the tests to mock this functionality more easily, since it doesn't need to capture calls to `InetAddress`, just itself

Reply:
ahh ok yeah missed that!